### PR TITLE
Improve JSON handling

### LINKED ISSUE #122: Consider using JSON for Modern C++ in place of rapidjson
See https://github.com/nlohmann/json
Conan recipe: https://github.com/vthiery/conan-jsonformoderncpp

I think this would make a large difference to the maintainability of the codebase and is thus compatible with the aims for the final few months of BrightnESS.

### PR DESCRIPTION
**Pull request by dominikwerder**
### Description of work

- Replace usage of the `rapidjson` library by json for modern c++
- Replacements in source code, build setup, conanfiles, documentation

### Issue

Closes #122 

### Acceptance Criteria

Everything should work as before.

---

## Code Review (To be filled in by the reviewer only)

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel (does someone want to automate this?).

### CODE REVIEW

File: src/helper.h [hunk 1]
```diff
@@ -25,66 +25,7 @@ std::vector<char> binary_to_hex(char const *data, uint32_t len);
...
-get_json_ret_int get_int(rapidjson::Value const *v, std::string path);
-get_json_ret_uint get_uint(rapidjson::Value const *v, std::string path);
-get_json_ret_bool get_bool(rapidjson::Value const *v, std::string path);
-get_json_ret_array get_array(rapidjson::Value const &v, std::string path);
-get_json_ret_object get_object(rapidjson::Value const &v, std::string path);
-
-std::string pretty_print(rapidjson::Document const *v);
-
 template <typename T>
 std::pair<short int, T> to_num(const std::string &string) {
```
Comment (matthew-d-jones) [inline src/helper.h]: I think `sleep_ms()` and all the `to_num()` overloads can also be removed. They don't seem to be used anywhere.

File: src/schemas/ev42/ev42_rw.cpp [hunk 2]
```diff
@@ -42,35 +44,72 @@ uint64_t FlatbufferReader::timestamp(Msg const &msg) const {
...
+  auto ConfigurationStreamJson = json::parse(ConfigurationStream);
+  try {
+    index_every_bytes =
+        ConfigurationStreamJson["nexus"]["indices"]["index_every_kb"]
+            .get<uint64_t>() *
+        1024;
     LOG(Sev::Debug, "index_every_bytes: {}", index_every_bytes);
-  } else if (auto x = get_int(&config_stream, "nexus.indices.index_every_mb")) {
-    index_every_bytes = uint64_t(x.v * 1024 * 1024);
+  } catch (...) { /* it's ok if not found */
```
Comment (matthew-d-jones) [inline src/schemas/ev42/ev42_rw.cpp]: Can't we catch anything more specific?

File: src/tests/IntegrationTests.cpp [hunk 3]
```diff
@@ -1,22 +1,5 @@
...
-  CLI::App App{""};
-  auto Options = std::unique_ptr<MainOpt>(new MainOpt());
-  setCLIOptions(App, *Options);
-
-  CLI11_PARSE(App, argc, argv);
-  setup_logger_from_options(*Options);
-  SetTestOptions(Options.get());
-
-  return RUN_ALL_TESTS();
+  // Should be no longer needed.
```
Comment (matthew-d-jones) [inline src/tests/IntegrationTests.cpp]: Agreed, any reason not to remove this file?

File: src/schemas/ev42/ev42_rw.cpp (hunk 2 above)
Comment (dominikwerder) [reply src/schemas/ev42/ev42_rw.cpp]: Needs a bit more time, created ticket https://jira.esss.lu.se/browse/DM-925

File: src/tests/IntegrationTests.cpp (hunk 3 above)
Comment (dominikwerder) [reply src/tests/IntegrationTests.cpp]: :+1:

**Pull request merged**